modules/serverless-containersource-ossm.adoc

```diff
@@ -0,0 +1,128 @@
+// Module included in the following assemblies:
+//
+// * /serverless/eventing/event-sources/serverless-custom-event-sources.adoc
+
+:_content-type: PROCEDURE
+[id="serverless-containersource-ossm_{context}"]
+= Integrating {SMProductShortName} with ContainerSource
+
+.Prerequisites
+
+// TODO link to ../serverless/integrations/serverless-ossm-setup.adoc#serverless-ossm-setup_serverless-ossm-setup
+
+* You have integrated {SMProductShortName} with {ServerlessProductName}.
+
+.Procedure
+
+. Create a `Service` in a namespace that is member of the `ServiceMeshMemberRoll`.
++
+[source,yaml]
+----
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: event-display
+  namespace: <namespace> <1>
+spec:
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "true" <2>
+        sidecar.istio.io/rewriteAppHTTPProbers: "true"
+    spec:
+      containers:
+      - image: quay.io/openshift-knative/knative-eventing-sources-event-display:latest
+----
+<1> A namespace is member of the `ServiceMeshMemberRoll`.
+<2> Injects {SMProductShortName} sidecars into the Knative service pods.
+
+. Apply the `Service` resource.
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+. Create a `ContainerSource` in a namespace that is member of the `ServiceMeshMemberRoll` and sink set to the `event-display`.
++
+[source,yaml]
+----
+apiVersion: sources.knative.dev/v1
+kind: ContainerSource
+metadata:
+  name: test-heartbeats
+  namespace: <namespace> <1>
+spec:
+  template:
+    metadata: <2>
+      annotations:
+        sidecar.istio.io/inject": "true"
+        sidecar.istio.io/rewriteAppHTTPProbers: "true"
+    spec:
+      containers:
+        # This corresponds to a heartbeats image URI that you have built and published
+        - image: gcr.io/knative-releases/knative.dev/eventing/cmd/heartbeats
+          name: heartbeats
+          args:
+            - --period=1s
+          env:
+            - name: POD_NAME
+              value: "example-pod"
+            - name: POD_NAMESPACE
+              value: "event-test"
+  sink:
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: event-display-service
```

Comment:
```suggestion
      name: event-display
```

(to match the name of the service created above)

Reply:
Done!